### DISCUSSION TITLE
nextjs integration

### DISCUSSION THREAD

**Discussion by MehmetBozkir** [TanStack/query#6985] [General]
Is there any video or documentation on how to use nextjs?

**Comment by truexor** [comment]
TL;DR - There is a detailed [NextJS documentation](https://nextjs.org/docs) out there
TL;DR - This [Medium blog](https://medium.com/@ahmed.nums345/a-comprehensive-guide-to-next-js-5f3b03b49def) does the same, and I prefer the same

<hr/>

## Full guide

### Prerequisites:
- `node` and `npm` is installed
- have a directory created where nextjs project is to be used (call it `/abc`)
- open CLI and cd into the folder path 

### Run:
- `npm create next-app@latest`
- go through the options as required <br/>
![image](https://github.com/TanStack/query/assets/96081625/0dc585ab-cdd7-4bbd-8a3c-30f57c1038de)

- once initialized, run `npm run dev` and you will be good to go

**Comment by zainytech** [comment]
You can go through these pdfs to get started with nextjs...
[why-choose-next-.pdf](https://github.com/TanStack/query/files/14996406/react-to-next-post1.pdf)
[how-to-create-a-nextjs-project.pdf](https://github.com/TanStack/query/files/14996373/how-to-create-a-nextjs-project.pdf)
[nextjs-components.pdf](https://github.com/TanStack/query/files/14996412/nextjs-components.pdf)
[app-route-in-nextjs.pdf](https://github.com/TanStack/query/files/14996414/app-rotes-in-nextjs.pdf)

In addition to the information I've provided, I also have PDF resources on data fetching and creating API routes in Next.js. I would be happy to share them with you if you're interested.👍
